server/src/main/java/org/elasticsearch/index/query/QueryRewriteContext.java

```diff
@@ -10,28 +10,68 @@
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.client.internal.Client;
 import org.elasticsearch.common.util.concurrent.CountDown;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.analysis.IndexAnalyzers;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.MapperBuilderContext;
+import org.elasticsearch.index.mapper.MapperService;
+import org.elasticsearch.index.mapper.MappingLookup;
+import org.elasticsearch.index.mapper.TextFieldMapper;
 import org.elasticsearch.xcontent.XContentParser;
 import org.elasticsearch.xcontent.XContentParserConfiguration;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.Map;
 import java.util.function.BiConsumer;
 import java.util.function.LongSupplier;
+import java.util.function.Predicate;
 
 /**
  * Context object used to rewrite {@link QueryBuilder} instances into simplified version.
  */
 public class QueryRewriteContext {
+    protected final MapperService mapperService;
+    protected final MappingLookup mappingLookup;
+    protected final Map<String, MappedFieldType> runtimeMappings;
+    protected final IndexSettings indexSettings;
+    protected final Index fullyQualifiedIndex;
     private final XContentParserConfiguration parserConfiguration;
     protected final Client client;
     protected final LongSupplier nowInMillis;
     private final List<BiConsumer<Client, ActionListener<?>>> asyncActions = new ArrayList<>();
+    protected boolean allowUnmappedFields;
+    protected boolean mapUnmappedFieldAsString;
+    protected Predicate<String> allowedFields;
 
-    public QueryRewriteContext(XContentParserConfiguration parserConfiguration, Client client, LongSupplier nowInMillis) {
+    public QueryRewriteContext(
```

Comment:
Is the idea to create a new instance of this class with mapper service etc. in #96161?

Reply:
Hi Martijn...the idea here is to just move stuff and later on for the other PR just:
* remove unnecessary things that we have here as a result of moving fields into the `QueryRewriteContext`
* include the rewrite in the SearchService so to actually trigger the rewrite in the data node
* eventually change the code for at least `MatchPhaseQueryBuilder` to actually implement the shard skipping logic for the query used by Observability integrations.

Not sure why we need to create a new class here.

Reply:
The main difference will be in creating a `QueryRewriteContext` where in the other PR we create a `SearchExecutionContext`. This way we will not need to set the `IndexSearcher` to null, which was one of the issues there.